### DISCUSSION TITLE
Get roles in citadel project via token introspection / userinfo

### DISCUSSION THREAD

**Discussion by errnesto** [zitadel/zitadel#6129] [Q&A] [answered]
Is it possible to include the roles of the user in the token so I can get them via the introspection or userinfo endpoint?

I would like to figure out if the current user is able to access the zitadel management api.
I tried to add `urn:zitadel:iam:org:project:{ZITADEL_PROJECT_ID}:roles` to the scope of my token but without success.

I also added the `urn:zitadel:iam:org:project:id:zitadel:aud` claim which will add the zidatel project to the audince.
But of course simply having the audience is not proove that I'm able to actually manage users.

I think I should be able to solve this by using https://zitadel.com/docs/apis/resources/auth/auth-service-list-my-zitadel-permissions but I was wondering if I can safe one request here…

**Comment by hifabienne** [comment, marked as answer]
Hei @errnesto 
There are two different use cases on roles in ZITADEL. One is if you have your own projects with roles, and want to include those roles in the token. This is possible with the settings on the project "Assert Roles on Authentication" and on the application "User roles inside ID Token"
<img width="695" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/40bc678c-7a50-4662-a858-4253305b788a">
<img width="695" alt="image" src="https://github.com/zitadel/zitadel/assets/38692350/eab8c556-748f-42cc-a6d3-1c2b30c83c39">

The second use case is the ZITADEL "internal roles" or how we call it the manager roles. Like Org_OWNER, IAM_OWNER, etc.
From your text I assume that you want to include the manager roles right?
At the moment  it is not possible to include those directly in the token, and you need the API mentioned above.

**Reply by mffap** [reply]
@dakshitha we should consider this in our guide

**Comment by errnesto** [comment]
Hey thank you for the very quick respose!

Yes I am interested in the manager roles. So I guess I have to fall back to the api then :-)
(Which is also fine!)

Again thank you for your help :-)
